gcp/modules/gpii-flowmanager/dns.tf

```diff
@@ -0,0 +1,16 @@
+data "kubernetes_service" "istio-ingressgateway" {
+  metadata {
+    name      = "istio-ingressgateway"
+    namespace = "istio-system"
+  }
+}
+
+resource "google_dns_record_set" "flowmanager-dns" {
+  name         = "flowmanager.${var.domain_name}."
```

Comment:
So currently the `*.` wildcard would be kept pointing to the `nginx-ingress` load-balancer, and I was thinking about getting rid of it in the cleanup phase.

I think we can keep the wildcard around if we want to, we would still need to rewrite the code (currently it's handled by exekube module, which needs to be created before whole k8s cluster is deployed, but we don't know the actual IP till we deploy Istio - afaik there's no way to use Istio with static external IP atm. in an automated way)

So the IP will have to change in any case.

Do you prefer to have one `*.` wildcard instead of individual records in general? I'm more of a fan of the latter, as it gives you better granularity, but otherwise that should be possible.

Re. DNS challenge - neither way should affect this, the DNS challenge always needs to create `_acme-challenge.<domain>` TXT records.

Reply:
> Do you prefer to have one *. wildcard instead of individual records in general?

All things being equal, yes. My experience is that the fewer DNS updates we make, and thus the fewer changes that have to propagate through multiple caches to reach a developer, the better.

Reply:
> All things being equal, yes. 

I was looking into this a bit more and it would complicate the migration process quite a bit (as mentioned, because of the ordering the code has to be moved out of exekube module, and the IP will change). And rollback would be also more complex.

> My experience is that the fewer DNS updates we make, and thus the fewer changes that have to propagate through multiple caches to reach a developer, the better.

So this is the reason why this change is done in a separate step.
- We can deploy everything and take as much time as we want to test (Phase 1 & 2). 
- Change the DNS (This PR)
- Wait as long as we want before we decommission the old ingress via nginx. - it will work in parallel with istio. This should give everything enough time to propagate eve if something does not respect TTLs.
- Continue with the rest... (Phase 4+)